demisto_sdk/commands/validate/validate_manager_v2.py

```diff
@@ -0,0 +1,138 @@
+        staged=False,
+        allow_autofix=False,
+        config_file_path=None,
+        only_committed_files=None,
+        ignore_support_level=False,
+    ):
+        self.ignore_support_level = ignore_support_level
+        self.validate_all = validate_all
+        self.file_path = file_path
+        self.staged = staged
+        self.run_with_multiprocessing = multiprocessing
+        self.allow_autofix = allow_autofix
+        self.config_file_path = config_file_path
+        self.category_to_run = config_file_category_to_run
+        self.json_file_path = json_file_path
+        self.validation_results = ValidationResults(json_file_path=self.json_file_path)
+        self.validate_graph = False
+        self.config_reader = ConfigReader(
+            config_file_path=self.config_file_path,
+            category_to_run=self.category_to_run,
+        )
+        self.initializer = Initializer(
+            use_git=use_git,
+            staged=self.staged,
+            committed_only=only_committed_files,
+            prev_ver=prev_ver,
+            file_path=self.file_path,
+            all_files=self.validate_all,
+        )
+        self.objects_to_run: Set[
+            Tuple[BaseContent, Optional[BaseContent]]
+        ] = self.initializer.gather_objects_to_run()
+        self.use_git = self.initializer.use_git
+        self.committed_only = self.initializer.committed_only
+        (
+            self.validations_to_run,
+            self.warnings,
+            self.ignorable_errors,
+            self.support_level_dict,
+        ) = self.config_reader.gather_validations_to_run(
+            use_git=self.use_git, ignore_support_level=self.ignore_support_level
+        )
+        self.validate_graph = False
+        self.validators = self.filter_validators()
+        if self.validate_graph:
+            logger.info("Graph validations were selected, will init graph")
+            self.init_graph()
+
+    def run_validation(self) -> int:
+        """
+            Running all the relevant validation on all the filtered files based on the should_run calculations,
+            calling the fix method if the validation fail, has an autofix, and the allow_autofix flag is given,
+            and calling the post_results at the end.
+        Returns:
+            int: the exit code to obtained from the calculations of post_results.
+        """
+        for validator in self.validators:
+            if filtered_content_objects_for_validator := list(
+                filter(
+                    lambda content_object: validator.should_run(
+                        content_object[0],
+                        self.ignorable_errors,
+                        self.support_level_dict,
+                    ),
+                    self.objects_to_run,
+                )
+            ):
+                validation_results: List[ValidationResult] = validator.is_valid(*zip(*filtered_content_objects_for_validator))  # type: ignore
```

Comment:
Because in the end, we want to run the validation against the old files only in some cases, so it's easier to split it into two lists and run only on the content_objects list when there's no need for the second list.